### PR TITLE
set up associations

### PR DESCRIPTION
**Pull request by yohannesdagnachew**
### Description
In this project, I created the remaining classes for the OOP school library and created the associations between them.
- Created a class Classroom with the following ✅ 
- Created the has-many/belongs-to relationship between Classroom and Students ✅ 
-  Created a class Book ✅ 
- Created a class Rental ✅ 
- Created the many-to-many relationship between Person and Book using the intermediate class Rental ✅

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline book.rb]: - [ ] In your book class there should be a method for adding rentals, kindly fix that
 ```
 def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
   ....
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
+
+  def initialize(date)
```
Comment (VICTOR-CHUKWU) [inline rental.rb]: - [ ] your constructor function should take book and person as paramters
```ruby
 def initialize(date, book, person)
    .
```

File: person.rb [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
   end
 
-  private
-
   def age?
     @age > 18
   end
+
+  private :age?
 end
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [ ] There should be a method for adding rentals, kindly add that
```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
   ....
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @yohannesdagnachew  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @yohannesdagnachew,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**